packages/teleterm/src/ui/services/usage/usageService.ts

```diff
@@ -0,0 +1,192 @@
+        os: platform,
+        osVersion,
+        connectVersion,
+      },
+    });
+  }
+
+  captureProtocolRun(
+    uri: ClusterOrResourceUri,
+    protocol: 'ssh' | 'kube' | 'db'
+  ): Promise<void> {
+    const clusterProperties = this.getClusterProperties(uri);
+    if (!clusterProperties) {
+      return;
+    }
+    return this.reportEvent(clusterProperties.authClusterId, {
+      protocolUse: {
+        clusterName: clusterProperties.clusterName,
+        userName: clusterProperties.userName,
+        protocol,
+      },
+    });
+  }
+
+  captureAccessRequestCreate(
+    uri: ClusterOrResourceUri,
+    kind: 'role' | 'resource'
+  ): Promise<void> {
+    const clusterProperties = this.getClusterProperties(uri);
+    if (!clusterProperties) {
+      return;
+    }
+    return this.reportEvent(clusterProperties.authClusterId, {
+      accessRequestCreate: {
+        clusterName: clusterProperties.clusterName,
+        userName: clusterProperties.userName,
+        kind,
+      },
+    });
+  }
+
+  captureAccessRequestReview(uri: ClusterOrResourceUri): Promise<void> {
+    const clusterProperties = this.getClusterProperties(uri);
+    if (!clusterProperties) {
+      return;
+    }
+    return this.reportEvent(clusterProperties.authClusterId, {
+      accessRequestReview: {
+        clusterName: clusterProperties.clusterName,
+        userName: clusterProperties.userName,
+      },
+    });
+  }
+
+  captureAccessRequestAssumeRole(uri: ClusterOrResourceUri): Promise<void> {
+    const clusterProperties = this.getClusterProperties(uri);
+    if (!clusterProperties) {
+      return;
+    }
+    return this.reportEvent(clusterProperties.authClusterId, {
+      accessRequestAssumeRole: {
+        clusterName: clusterProperties.clusterName,
+        userName: clusterProperties.userName,
+      },
+    });
+  }
+
+  captureFileTransferRun(
+    uri: ClusterOrResourceUri,
+    direction: 'download' | 'upload'
+  ): Promise<void> {
+    const clusterProperties = this.getClusterProperties(uri);
+    if (!clusterProperties) {
+      return;
+    }
+    return this.reportEvent(clusterProperties.authClusterId, {
+      fileTransferRun: {
+        clusterName: clusterProperties.clusterName,
+        userName: clusterProperties.userName,
+        direction,
+      },
+    });
+  }
+
+  captureUserJobRoleUpdate(jobRole: string): Promise<void> {
+    return this.reportNonAnonymizedEvent({
+      userJobRoleUpdate: {
+        jobRole,
+      },
+    });
+  }
+
+  private reportNonAnonymizedEvent(
+    prehogEventReq: PrehogEventReq
+  ): Promise<void> {
+    return this.reportEvent('', prehogEventReq);
+  }
+
+  private reportEvent(
+    authClusterId: string,
+    prehogEventReq: PrehogEventReq
+  ): Promise<void> {
+    const isCollectingUsageMetricsEnabled = this.configService.get(
+      'usageMetrics.enabled'
+    ).value;
+
+    if (!isCollectingUsageMetricsEnabled) {
+      return;
+    }
+
+    return this.tshClient.reportUsageEvent({
```

Comment:
Good point, I wasn't sure what to do about it, so I left it without handling (so we would be able to at least see it in dev mode). As you said an error is only shown when there is a problem between Electron and tshd, not because prehog is down for example (this problem will be visible in `tshd` logs only).
Adding a warning looks like the best option to handle this.